### PR TITLE
Feature/42 currency pairs periodical

### LINKED ISSUE #37: Processor module implementation
Implement module processing data obtained from Fetcher module to create statistical analysis.

### LINKED ISSUE #42: Requirement implementation - periodical currency pair variance
Monthly and quarter variance distribution for currency pair (e.g. EUR/USD) chosen by user. Analysis is based on daily changes in value from previous day and output it histogramatically.

### PR DESCRIPTION
**Pull request by mczaj224798**
resolves #42
resolves #37 
resolves #62

### CODE REVIEW

**Review by MavoSV8 — APPROVED** [review]
ok

File: Project2/src/libs/application.py [hunk 1]
```diff
@@ -1,6 +1,52 @@
...
+
+        if data is None:
+            print("error: couldn't fetch quotas for currency: {}".
+                  format(self.currency))
+            return 1
+
+        self.proc.set_data(data)
+
+        if self.mode == "DA":
+            data2 = self.fetcher.fetch(self.time, self.currency2)
```
Comment (mbarancewicz) [inline Project2/src/libs/application.py:31]: repeating yourself (lines 23-26), maybe extract to method

File: Project2/src/libs/processor.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    def set_data2(self, data):
+        if isinstance(data, dict) and len(data) > 1:
+            self.data2 = data
+
+    def set_mode(self, mode):
+        result = mode in self.MODES
+
+        if result:
+            self.mode = mode
+        return result
```
Comment (mbarancewicz) [inline Project2/src/libs/processor.py:73]: do we need to return result at all? if not then result variable is redundant and could be handled like:
if mode in self.MODES:
    self.mode = mode

File: Project2/src/libs/application.py (hunk 1 above)
Comment (mczaj224798) [reply Project2/src/libs/application.py:31]: Moving this to method would only decrease readability.  Its a simple null check for output of the fetcher. No point for creating method that would also require return validation.

File: Project2/src/libs/cli_parser.py [hunk 3]
```diff
@@ -1,51 +1,52 @@
...
-
-        self.parser.add_argument("-c2", "--currency2", help="second currency for DA",
-            choices=["THB", "USD", "AUD", "HKD", "CAD", "NZD", "SGD", "EUR"
-                "HUF", "CHF", "GBP", "UAH", "JPY", "CZK", "DKK", "ISK", "NOK", "SEK", "HRK",
-                "RON", "BGN", "TRY", "ILS", "CLP", "PHP", "MXN", "ZAR", "BRL", "MYR", "RUB",	
-                "IDR", "INR", "KRW", "CNY", "XDR"],
-            required=False)
+                                 choices=["CA", "SDA", "DA"],
+                                 help="type of analysis, [CA] - Change Analysis,\
+                                 [SDA] - Statistical Data Analysys, [DA] - \
```
[the remark this answers is not in the document]
Comment (mczaj224798) [reply Project2/src/libs/cli_parser.py:29]: Minor, I will fix it when integrating.

File: Project2/src/libs/processor.py (hunk 2 above)
Comment (mczaj224798) [reply Project2/src/libs/processor.py:73]: You need to inform the user wheter the operation was successful. If not, he has to act accordingly.

File: Project2/src/libs/application.py (hunk 1 above)
Comment (mbarancewicz) [reply Project2/src/libs/application.py:31]: fine, its pretty concise

**Review by mbarancewicz — APPROVED** [review]

**Comment by mbarancewicz** [comment]
approved

**Pull request merged**